bpo-22377: Fixes documentation for %Z in datetime

karlcow opened the pull request:
<!--
Thanks for your contribution!
Please read this comment in its entirety. It's quite important.

# Pull Request title

It should be in the following format:

```
bpo-NNNN: Summary of the changes made
```

Where: bpo-NNNN refers to the issue number in the https://bugs.python.org.

Most PRs will require an issue number. Trivial changes, like fixing a typo, do not need an issue.

# Backport Pull Request title

If this is a backport PR (PR made against branches other than `master`),
please ensure that the PR title is in the following format:

```
[X.Y] <title from the original PR> (GH-NNNN)
```

Where: [X.Y] is the branch name, e.g. [3.6].

GH-NNNN refers to the PR number from `master`.

-->

This fixes the issue discussed in https://bugs.python.org/issue22377
and fixes it according to the comments made by Paul Ganssle @pganssle

* It clarifies which values are acceptable in the table
* It extends the note with a clearer information on the valid values

<!-- issue-number: [bpo-22377](https://bugs.python.org/issue22377) -->
https://bugs.python.org/issue22377
<!-- /issue-number -->


Automerge-Triggered-By: @pganssle

karlcow committed:
bpo-22377: Fixes documentation for %Z in datetime

blurb-it[bot] committed:
📜🤖 Added by blurb_it.

pganssle commented on Doc/library/datetime.rst:
```diff
@@ -2537,6 +2537,12 @@ Notes:
       string. Otherwise ``%Z`` is replaced by the returned value, which must
       be a string.
 
+      Note that ``%Z`` accepts only the hardcoded values: ``UTC`` and ``GMT``,
```
Perhaps we should specify that this is for `strptime`? "Note that `strptime` only accepts certain values for `%Z`: ..."

karlcow replied:
Thanks good suggestion.

pganssle commented on Misc/NEWS.d/next/Documentation/2019-10-01-10-53-46.bpo-22377.5ni-iC.rst:
```diff
@@ -0,0 +1 @@
+Improves ``%Z`` in datetime documentation.
```
```suggestion
Improves documentation of the values that :meth:`datetime.datetime.strptime` accepts for ``%Z``.
Patch by Karl Dubost.
```

Added a bit more detail to the changelog here, and also added your name. If you don't want to be credited, feel free to omit the "Patch by.." portion.

karlcow replied:
thx

pganssle reviewed (changes requested):
Overall this looks great, thanks @karlcow!

I just have two minor nits.

karlcow committed:
Update Misc/NEWS.d/next/Documentation/2019-10-01-10-53-46.bpo-22377.5ni-iC.rst

Co-Authored-By: Paul Ganssle <p.ganssle@gmail.com>

pganssle commented on Doc/library/datetime.rst:
```diff
@@ -2537,6 +2537,12 @@ Notes:
       string. Otherwise ``%Z`` is replaced by the returned value, which must
       be a string.
 
+      Note that ``strptime`` only accepts certain values for ``%Z``: ``UTC``
```
I think this wording would be better (the extra "and" is throwing me off here):

```
  Note that ``strptime`` only accepts certain values for ``%Z``: any value
  in ``time.tzname`` for your machine's locale and the hard-coded values
  ``UTC`` and ``GMT``. So someone living in Japan may have ``JST``, ``UTC``,
  and ``GMT`` as valid values, but probably not ``EST``. 
```

karlcow replied:
@pganssle no problem. Do you want me to rebase the pull request? with only one commit?

karlcow replied:
did you also scrap `It will return a ``ValueError`` for any invalid strings.` on purpose?

pganssle replied:
No, I just meant that there doesn't need to be any changes from "It will return a ``ValueError`` for any invalid strings." on.

karlcow replied:
OK I added it back. We are good to go.

karlcow committed:
bpo-22377: Adds ValueError information for %Z in datetime

pganssle committed:
Minor tweaks to %Z documentation.

- Made the more unusual value more prominent.
- Made a clearer separation between the `strptime` and `strftime` parts.

pganssle reviewed (approved):
I am so sorry it's taken me so long to swing back around to this @karlcow.

I made some minor tweaks here and everything looks good. Please take a look at my changes and let me know if you agree with my changes. If so, I'm happy to merge.

(By the way, I did a force-push because I rebased against master, if you need to make more changes, please do a force-pull or something equivalent first).
